google/cloud/aiplatform/datasets/dataset.py

```diff
@@ -52,70 +58,95 @@ def __init__(
         """Retrieves an existing managed dataset given a dataset name or ID.
 
         Args:
-            dataset_name (str):
+            dataset_name: (str)
                 Required. A fully-qualified dataset resource name or dataset ID.
                 Example: "projects/123/locations/us-central1/datasets/456" or
                 "456" when project and location are initialized or passed.
-            project (str):
+            project: (str) = None
                 Optional project to retrieve dataset from. If not set, project
                 set in aiplatform.init will be used.
-            location (str):
+            location: (str) = None
                 Optional location to retrieve dataset from. If not set, location
                 set in aiplatform.init will be used.
-            credentials: Optional[auth_credentials.Credentials]=None,
+            credentials: Optional[auth_credentials.Credentials] = None,
                 Custom credentials to use to upload this model. Overrides
                 credentials set in aiplatform.init.
+
+        Raises:
+            ValueError if the dataset type of the retrieved dataset resource is
+            not supported by the 'Dataset' class.
         """
 
         super().__init__(project=project, location=location, credentials=credentials)
         self._gca_resource = self._get_gca_resource(resource_name=dataset_name)
+        self._validate_metadata_schema_uri()
+
+    @property
+    def metadata_schema_uri(self) -> str:
+        """The metadata schema uri of this dataset resource."""
+        return self._gca_resource.metadata_schema_uri
+
+    def _validate_metadata_schema_uri(self) -> bool:
```

Comment:
I guess my comment disappeared as things got moved, but I was saying that I wasn't a fan of having the `_support_import_schema_uris`, `_support_import_schema_uris` and the validation logic in `dataset.py`.

This is knowledge that the subclasses themselves should have.

e.g. Only ImageDataset needs to know that certain import_schema_uris are applicable for Images.

I don't think it's applicable to a generic Dataset object, so it feels strange for it to live here.

Reply:
As discussed offline, the validation of import_schema_uri will be revisit later.